crates/sui-adapter/src/execution_engine.rs

```diff
@@ -87,14 +87,55 @@ pub fn execute_transaction_to_effects<
         protocol_config,
     );
 
-    let (status, execution_result) = match execution_result {
-        Ok(results) => (ExecutionStatus::Success, Ok(results)),
-        Err(error) => {
-            let (status, command) = error.to_execution_status();
-            (ExecutionStatus::new_failure(status, command), Err(error))
-        }
+    let status = if let Err(error) = &execution_result {
+        // Elaborate errors in logs if they are unexpected or their status is terse.
+        use ExecutionErrorKind as K;
+        match error.kind() {
+            K::InvariantViolation |
+            K::VMInvariantViolation => {
+                #[skip_checked_arithmetic]
+                tracing::error!(
+                    kind = ?error.kind(),
+                    tx_digest = ?transaction_digest,
+                    "INVARIANT VIOLATION! Source: {:?}",
+                    error.source(),
+                );
+            },
+
+            K::SuiMoveVerificationError |
+            K::VMVerificationOrDeserializationError => {
+                #[skip_checked_arithmetic]
+                tracing::info!(
+                    kind = ?error.kind(),
+                    tx_digest = ?transaction_digest,
+                    "VERIFICATION ERROR! Source: {:?}",
```

Comment:
Sometimes invariant violation errors that happen during publish end up getting double wrapped as a verification error, so we won't get to reproduce them by running the verifier locally, and we don't get the actual verification error in the transaction effects, so this is a way to get some early signal on the error.

I don't expect to see these errors often because  most people use our toolchain and our toolchain should be verifying that these errors don't happen before submitting the transaction, so if we do see them, it's usually a sign of a bug or something fishy going on.

Unlike the invariant violations, which I trace using `error!` I've traced these using `info!` to indicate the appropriate level of surprise...

Reply:
It's possible you meant loudness as in the fact that these are in all-caps?  I just did that for consistency with the invariant violation which definitely should be loud, but I can change it.

Reply:
I just see all caps and think "Oh no what's happened", but maybe that's just me

Reply:
I mean, given the last week, I think we all know what's happened... But I'll uncapitalise the ones other than invariant violation!